### ISSUE TITLE
Failed to connect to registry on the remote when the domain is proxied locally

### ISSUE THREAD

**Issue by imba-tjd** [microsoft/vscode-docker#1648]
## Environment

Host OS: Windows 10 20H1 (preview)
Server: Debian 11 bullseye (testing)
VSC: 1.42.1 (latest)
Docker Extension: 0.10 (latest)
Remote (SSH) Extension: 0.49 (latest)

## The Issue

I have set up a usable socks proxy listening on 127.0.0.1:1080, and I added `docker.com` to pac.txt so that visiting `*.docker.com` will be proxied.

When I try to connect to docker hub registry **on the remote**, I got a `Error: connect ECONNREFUSED 127.0.0.1:1080`.

![图片](https://user-images.githubusercontent.com/24759802/75158995-3c5ec500-5752-11ea-8170-8ca46862d05d.png)

Without remote or disable pac, I can connect to it successfully.

**Comment by ravipal** [comment]
@imba-tjd, can you please try to login to the registry from command line and see if that works?

**Comment by imba-tjd** [comment]
On my host OS: I didn't install docker toolbox, so I can't use Docker CLI. But this VSC docker extension **can** connect to registry even other functions say `Failed to connect. Is Docker installed and running?`

On the server: I can login via docker CLI.

**Comment by imba-tjd** [comment]
I found an easier way to reproduce.

1. On Windows, setup a global proxy to a non-exist address. For example `0.0.0.0:12345`

    ![图片](https://user-images.githubusercontent.com/24759802/75414491-ad28fb80-5963-11ea-827b-b0c38bb53bcf.png)

2. Open this extension. You should see an **expected** error

    ![图片](https://user-images.githubusercontent.com/24759802/75414580-f8dba500-5963-11ea-8d87-0850f5987c8d.png)

3. Use Remote - SSH to connect to the server and open this extension on the remote (SSH won't obey the "global proxy" in 1 so you can connect successfully. This is expected).
    **You will got the same error as 2**. This is not expected.

Note: I don't know whether is issue belongs to `Remote - SSH`. During my limited usage, I only see this issue on vscode-docker.

**Comment by bwateratmsft** [comment]
Ahh, I understand what's happening here. By default the Docker extension runs as a UI extension--what this means is that, when connecting with the Remote tools, it's running on the client side rather than the server side. You can change that by installing it in the SSH session:

1. Add this to your settings.json:
```
    "remote.extensionKind": {
        "ms-azuretools.vscode-docker": "workspace"
    }
```
2. Connect to the remote session
3. In the Extensions install view, install Docker in the remote SSH session:
![image](https://user-images.githubusercontent.com/36966225/75455880-d8055680-5947-11ea-9654-a852b50916a9.png)

Once you've done that, it should stop trying to use the proxy settings from the local machine while in the Remote - SSH session.

**Comment by bwateratmsft** [comment]
Let us know if that works, thanks!

**Comment by bwateratmsft** [comment]
Let us know if that works, thanks!

**Comment by imba-tjd** [comment]
Unfortunately it doesn't work. 

![图片](https://user-images.githubusercontent.com/24759802/75500041-d9e21f00-5a06-11ea-99b2-3f16b47d75ab.png)

![图片](https://user-images.githubusercontent.com/24759802/75500290-7a384380-5a07-11ea-9dc9-f6a89bd095ea.png)

**Comment by bwateratmsft** [comment]
Strange, we'll have to investigate why this is happening.

**Comment by bwateratmsft** [comment]
Just to make sure I correctly understand your original scenario--

1. Docker daemon is running on a remote machine
1. Remote - SSH is used to connect VSCode to the remote machine, and Docker extension is installed in that remote session
1. Proxy is running on and configured for the _local_ machine

And the error is that when using remote, the HTTP requests are trying to use proxy settings (from the local machine), but are running from the remote machine and therefore fail. Is my understanding correct?

**Comment by imba-tjd** [comment]
The previous one is correct.

**Comment by bwateratmsft** [comment]
@roblourens is this expected? We are using the [request](https://www.npmjs.com/package/request) library (which is deprecated and we need to use something else, but that's another issue). It seems like when connected via Remote - SSH, it is picking up proxy settings from the client side but executing the requests on the server side.

**Comment by roblourens** [comment]
@chrmarti?

**Comment by chrmarti** [comment]
Filed https://github.com/microsoft/vscode/issues/92064 to track on our side.

**Comment by bwateratmsft** [comment]
Thanks @chrmarti. I'll leave this item open as a tracking issue for now.

**Comment by bwateratmsft** [comment]
I have verified that this is now fixed in 1.44 which just released. I used @imba-tjd's steps from [this comment](https://github.com/microsoft/vscode-docker/issues/1648#issuecomment-591788899) above.

Local session (error, as expected):
![image](https://user-images.githubusercontent.com/36966225/78833025-35b2a700-79ba-11ea-885d-b0f5475bf2ca.png)

Remote SSH session (no error, because it's no longer erroneously trying to use the proxy settings from local):
![image](https://user-images.githubusercontent.com/36966225/78833092-524edf00-79ba-11ea-8373-3a22b74df1e9.png)
